### PR TITLE
Menu should be configurable and saved in database #339

### PR DESCRIPTION
**Pull request by sirnone**

### CODE REVIEW

**Comment by ximex** [comment]
Sooo. I have done a quick overview.
Things i found:
- unused imported globals
- unused declared variables
- some codestyle things
- at requires we should always use `__DIR__`
- SQL queries are not all prepared statements
- incomplete phpdocs
- more to come...

I will fix most of this
after fixing this i will look again more in detail

**Comment by Fasse** [comment]
@ximex can we merge?

**Comment by ximex** [comment]
@Fasse no. I will push some fixes today and i hope the rest tomorrow

**Comment by Fasse** [comment]
Ok, no problem

**Comment by ximex** [comment]
There were many `SELECT  *` added.
Wouldn't it good if we change all `SELECT  *` to `SELECT column_name, ...` to see exactly what columns we get and only return the columns we need?

**Comment by Fasse** [comment]
Yes, I would also prefer a `SELECT column_name` syntax.

**Comment by ximex** [comment]
more fixes are coming tomorrow.
But please someone fix the `$user` bug

**Comment by Fasse** [comment]
The $user bug ?!?

**Comment by ximex** [comment]
Look comment above. `$user` is not defined

**Comment by Fasse** [comment]
Which comment? I cant See a comment with $user!

**Comment by ximex** [comment]
Here: https://github.com/Admidio/admidio/pull/700/files/c0088b1d9b262d86dd76e4d3c2e83c774233ba8f#diff-c1ab13b4ccae594442e8f0e2e2f70c18R182

File: adm_program/system/classes/component.php [hunk 1]
```diff
@@ -89,4 +89,128 @@ public function checkDatabaseVersion()
...
+            case 'PHOTOS':
+                if((int) $gSettingsManager->get('enable_photo_module') === 1
+                || ((int) $gSettingsManager->get('enable_photo_module') === 2 && $gValidLogin))
+                {
+                    return true;
+                }
+                break;
+
+            case 'PROFILE':
+                if($gCurrentUser->hasRightViewProfile($user))
```
[the remark this answers is not in the document]
Comment (sirnone) [reply adm_program/system/classes/component.php]: case 'PROFILE' is not possible I think (this com_name_intern is  not in our Code)
can you confirm, that we can remove this case from the Code.

[the remark this answers is not in the document]
Comment (Fasse) [reply adm_program/system/classes/component.php]: fixed, change this to own profile because other profiles could not be checked.

**Comment by ximex** [comment]
The rest comes tomorrow. There are 4 files (`htmlpage.php`, `tablecategory.php, `tablemenu.php`, `menu_function.php`) left where i have more to fix/improve

**Comment by ximex** [comment]
@sirnone @Fasse i think there are two foreign keys missing
```sql
ALTER TABLE %PREFIX%_menu
    ADD CONSTRAINT %PREFIX%_fk_men_men_parent  FOREIGN KEY (men_men_id_parent)  REFERENCES %PREFIX%_menu (men_id)                ON DELETE SET NULL ON UPDATE RESTRICT,
    ADD CONSTRAINT %PREFIX%_fk_men_com_id      FOREIGN KEY (men_com_id)         REFERENCES %PREFIX%_categories (cat_id)          ON DELETE RESTRICT ON UPDATE RESTRICT;
```

**Comment by ximex** [comment]
I'm happy now -> could be merged. Maybe i find some more but that could be done later in master too

**Comment by Fasse** [comment]
@ximex  please test your changes:
> Fatal error: Uncaught Error: Cannot use object of type stdClass as array in /admidio/adm_program/system/classes/htmlpage.php:750 Stack trace: #0 /admidio/adm_program/index.php(64): HtmlPage->showMainMenu() #1 {main} thrown in /admidio/adm_program/system/classes/htmlpage.php on line 750

**Comment by ximex** [comment]
@Fasse hmmm don't know were this changes get lost...

**Comment by Fasse** [comment]
it's done :)

**Pull request merged**